portfolio.js

```diff
@@ -0,0 +1,149 @@
+    usedtechnology1: 'html',
+    usedtechnology2: 'css',
+    usedtechnology3: 'javascript',
+    projectsrc: '#',
+    githubsrc: '#',
+  },
+];
+
+const hamburger = document.querySelector('#hamburger');
+const closed = document.querySelector('.close');
+const navListItems = document.querySelectorAll('.nav-list-item');
+const navList = document.querySelector('#nav-list');
+
+const mobileMenu = () => {
+  navList.classList.toggle('active');
+  hamburger.classList.toggle('close');
+  closed.classList.toggle('displayshow');
+};
+
+navListItems.forEach((element) => {
+  element.addEventListener('click', mobileMenu);
+});
+
+hamburger.addEventListener('click', mobileMenu);
+closed.addEventListener('click', mobileMenu);
+
+const images = document.getElementById('project-img');
+const details = document.getElementById('newdetail');
+const projectsrcs = document.getElementById('projectsrc');
+const projectnames = document.getElementById('projectname');
+const subprojectnames = document.getElementById('subprojectname');
+const years = document.getElementById('year');
+const frontendBackendDev = document.getElementById('frontend/backend');
+const technology1 = document.getElementById('tech1');
+const technology2 = document.getElementById('tech2');
+const technology3 = document.getElementById('tech3');
+const githubsrcs = document.getElementById('githubsrc');
+const button1 = document.getElementById('seeproject1');
+const button2 = document.getElementById('seeproject2');
+const button3 = document.getElementById('seeproject3');
+const button4 = document.getElementById('seeproject4');
+const closeBtn = document.querySelector('.popUpClose');
+const modal = document.querySelector('.modal-overlay');
+
+let count = 0;
+
+const projectmodal = () => {
+  const project = projects[count];
+  images.src = project.image;
+  details.textContent = project.details;
+  projectsrcs.href = project.projectsrc;
+  projectnames.textContent = project.projectname;
+  subprojectnames.textContent = project.subprojectname;
+  years.textContent = project.year;
+  frontendBackendDev.textContent = project.projecttype;
+  githubsrcs.href = project.githubsrc;
+  technology1.textContent = project.usedtechnology1;
+  technology2.textContent = project.usedtechnology2;
+  technology3.textContent = project.usedtechnology3;
+};
+
+button1.addEventListener('click', () => {
+  count = 0;
+  projectmodal();
+  modal.classList.add('open-modal');
+  document.body.style.overflow = 'hidden';
+});
+
+button2.addEventListener('click', () => {
+  count = 1;
+  projectmodal();
+  modal.classList.add('open-modal');
+  document.body.style.overflow = 'hidden';
+});
+
+button3.addEventListener('click', () => {
+  count = 2;
+  projectmodal();
+  modal.classList.add('open-modal');
+  document.body.style.overflow = 'hidden';
+});
+
+button4.addEventListener('click', () => {
+  count = 3;
+  projectmodal();
+  modal.classList.add('open-modal');
+  document.body.style.overflow = 'hidden';
```

Comment:
THis should be implemented in a more dynamic way. If you notice you're only repeating yourself unnecessarily which goes against the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). Imagine you have 100, or maybe a thousand project to display, would you repeat the code up to a thousand places. That will become cumbersome and would lead to a code that cannot be easily maintained and will also lead to a large file. Please refactor this into just one function

Reply:
I did not understand how to get this done while doing this but  I undestand how I can now.

I have made necessary changes.
Thanks